pageserver/src/layered_repository.rs

```diff
@@ -1507,6 +1507,15 @@ impl LayeredTimeline {
                 PageReconstructResult::Continue(cont_lsn) => {
                     // Fetch base image / more WAL from the returned predecessor layer
                     if let Some((cont_layer, cont_lsn)) = self.get_layer_for_read(seg, cont_lsn)? {
+                        if cont_lsn == curr_lsn {
+                            // We landed on the same layer again. Shouldn't happen, but if it does,
+                            // don't get stuck in an infinite loop.
+                            bail!(
+                                "could not find predecessor layer of segment {} at {}",
+                                seg.rel,
+                                cont_lsn
+                            );
+                        }
```

Comment:
Would this be considered programmer error, and if so should it be a panic?

Reply:
If this happens, it's probably a bug somewhere in our code. -1 on panicing, that could cripple the server pretty badly. For example, if this happens in response to a GetPage request, panicing could poison a mutex and stop the tenant or timeline from working altogether. I think it's better to keep running.